community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -757,9 +765,12 @@ private void close( Header.Writer headerWriter ) throws IOException
 
         try
         {
-            // Perform a checkpoint before closing. If no changes has happened since last checkpoint,
-            // no new checkpoint will be created.
-            checkpoint( IOLimiter.unlimited(), headerWriter );
+            if ( health.isHealthy() )
```

Comment:
@MishaDemianenko It's just a very strange inconsistency in the index component, to bypass the usual pattern for a particular event. Also, it's much better to have the layer outside gb+tree, e.g. label scan store to log something more contextual, saying that "native label scan store didn't checkpoint blabla" instead of having a log statement saying that some gb+tree did something... what's that to the user? Weird I'd say. Think about logging as surface too.

Let's keep abstractions and try not to bypass for quick fixes. Please.